testgrid/specs/storage-migration.yaml

```diff
@@ -0,0 +1,767 @@
+  postUpgradeScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    sleep 120
+    download_file_from_deployment_and_compare "migration-test" "default" "./test.data" "/data/test.data"
+    pvc_uses_provisioner "migration-test" "default" "rook"
+    test_pull_image_from_registry
+- name: Migrate from Rook 1.0.4 to Rook 1.4+
+  flags: "yes"
+  cpu: 6
+  numPrimaryNodes: 1
+  numSecondaryNodes: 2
+  installerSpec:
+    kubernetes:
+      version: 1.19.x
+    containerd:
+      version: 1.6.x
+    flannel:
+      version: 0.21.x
+    registry:
+      version: 2.8.x
+    kotsadm:
+      version: latest
+    ekco:
+      version: latest
+    rook:
+      version: 1.0.4
+  upgradeSpec:
+    kubernetes:
+      version: 1.19.x
+    containerd:
+      version: 1.6.x
+    flannel:
+      version: 0.21.x
+    registry:
+      version: 2.8.x
+    kotsadm:
+      version: latest
+    ekco:
+      version: latest
+    rook:
+      version: 1.4.x
+  postInstallScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    create_deployment_with_mounted_volume "migration-test" "default" "/data"
+    create_random_file_and_upload_to_deployment "migration-test" "default" "./test.data" "/data/test.data"
+    test_push_image_to_registry
+  postUpgradeScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    sleep 120
+    download_file_from_deployment_and_compare "migration-test" "default" "./test.data" "/data/test.data"
+    pvc_uses_provisioner "migration-test" "default" "rook"
+    test_pull_image_from_registry
+- name: Migrate from Longhorn + Minio to Rook 1.4+
+  flags: "yes"
+  cpu: 6
+  numPrimaryNodes: 1
+  numSecondaryNodes: 2
+  installerSpec:
+    kubernetes:
+      version: 1.21.x
+    containerd:
+      version: 1.6.x
+    flannel:
+      version: 0.21.x
+    registry:
+      version: 2.8.x
+    kotsadm:
+      version: latest
+    ekco:
+      version: latest
+    longhorn:
+      version: 1.3.x
+    minio:
+      version: latest
+  upgradeSpec:
+    kubernetes:
+      version: 1.21.x
+    containerd:
+      version: 1.6.x
+    flannel:
+      version: 0.21.x
+    registry:
+      version: 2.8.x
+    kotsadm:
+      version: latest
+    ekco:
+      version: latest
+    rook:
+      version: 1.4.x
+  postInstallScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    create_deployment_with_mounted_volume "migration-test" "default" "/data"
+    create_random_file_and_upload_to_deployment "migration-test" "default" "./test.data" "/data/test.data"
+    test_push_image_to_registry
+  postUpgradeScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    sleep 120
+    download_file_from_deployment_and_compare "migration-test" "default" "./test.data" "/data/test.data"
+    pvc_uses_provisioner "migration-test" "default" "rook"
+    test_pull_image_from_registry
+    if kubectl get namespace/longhorn-system ; then
+      echo Longhorn namespace found after the upgrade.
+      exit 1
+    fi
+- name: Migrate from Rook 1.0.4 to OpenEBS + Minio
```

Comment:
Is not this one duplicated? 
I found twice `Migrate from Rook 1.0.4 to OpenEBS + Minio`

Reply:
Yeah, copy and pasta mistake. Good catch. Removed.